base/docs/basedocs.jl

```diff
@@ -2542,18 +2542,6 @@ Retrieve the declared type of the binding `name` from the module `module`.
 """
 Core.get_binding_type
 
-"""
-    Core.set_binding_type!(module::Module, name::Symbol, [type::Type])
-
-Set the declared type of the binding `name` in the module `module` to `type`. Error if the
-binding already has a type that is not equivalent to `type`. If the `type` argument is
-absent, set the binding type to `Any` if unset, but do not error.
-
-!!! compat "Julia 1.9"
-    This function requires Julia 1.9 or later.
-"""
-Core.set_binding_type!
```

Comment:
This is definitively a breaking change, as it removes a documented API present since v1.9, no?

Reply:
It's neither exported nor public - I think it is rightly considered an implementation detail of the syntax. In addition, I didn't find any public package that actually uses this any other way, so providing a deprecation implementation seems gratuitous.